apps/client/src/components/Questions.js

```diff
@@ -181,6 +203,16 @@ const Questions = ({
         // Check if currect question is causing a permit requirement
         const questionNeedsPermit = !!permitsPerQuestion[i];
 
+        // @TODO: Refactor this code and move to checker.js
+        // Bug fix in case of refresh: hide already future answered question (caused by setQuestionAnswers() in withChecker)
+        if (
+          contactConclusion &&
+          !checker._getUpcomingQuestions().length &&
+          questionIndex < i
+        ) {
+          return null;
+        }
```

Comment:
This was a bit hard to grasp. Can you add a comment like, 'We don't want to render future questions if the current index is the decisive answer for the Contact outcome'. If my assumption is wrong, please correct ;-)

Reply:
You're right. Adjusted